src/components/TreeReactFlow/layoutTree.ts

```diff
@@ -53,7 +59,25 @@ export const layoutTree = <
       const minY = Math.min(
         ...nodes.map((n) => n.position.y - (n.data.padding?.top || 0))
       );
-      const tree = treeMap(treeUnNormalized, (n) => ({
+      // Ensure that where a node is a right-child with no children of its own,
```

Comment:
As far as I can tell, this doesn't actually change anything. I've opened https://github.com/hackworthltd/primer-app/issues/1141 so we can look into it later, as it's not worth holding up this PR to fix.

Reply:
Never mind that, I'm going to split this commit out into a separate PR so that it can be evaluated independent of the first commit in this PR, which is what's required to work with the new backend API.